clippy_lints/src/methods/mod.rs

```diff
@@ -1307,6 +1307,32 @@ declare_clippy_lint! {
     "using `.iter().next()` on a sliced array, which can be shortened to just `.get()`"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Looks for unnecessary lazily evaluated closures on `Option` and `Result`.
+    ///
+    /// **Why is this bad?** Using eager evaluation is shorter and simpler in some cases.
```

Comment:
You could make a note here, that this is practically the inverse of [`or_fun_call`](https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call).

Reply:
Counterpart, perhaps, because this lint doesn't touch anything more complex than a member access, so applying it will not result in `or_fun_call` lints.

Thanks for the suggestion, I'll try to incorporate a note.

Reply:
> Counterpart

oh yeah, that would be the better word here.